pywren/pywren_ibm_cloud/wren.py

```diff
@@ -493,29 +491,31 @@ def clean(self, local_execution=True):
         storage_bucket = self.storage_config['storage_bucket']
         storage_prerix = self.storage_config['storage_prefix']
         storage_prerix = os.path.join(storage_prerix, self.executor_id)
-        
-        msg="Executor ID {} Cleaning partial results from PyWren bucket '{}'.".format(self.executor_id, storage_bucket)
-        logger.info(msg)
-        print(msg)
-        msg="Executor ID {} Cleaning partial results for '{}'.".format(self.executor_id, storage_prerix)
-        logger.info(msg)
-        print(msg)
 
+        msg = ("Executor ID {} Cleaning partial results from bucket '{}' "
+               "and prefix '{}'".format(self.executor_id, storage_bucket, storage_prerix))
+        logger.info(msg)
         if(logger.getEffectiveLevel() == logging.WARNING):
             print(msg)
 
         if local_execution:
-            #storage_config = json.dumps(self.storage_handler.get_storage_config())
-            #storage_config = storage_config.replace('"', '\\"')
-            clean_bucket(storage_bucket, storage_prerix, self.storage_config)
+            # 1st case: Not background. The main code waits until the cleaner finishes its execution.
+            # It is not ideal for performance tests, since it can take long time to complete.
+            #clean_os_bucket(storage_bucket, storage_prerix, self.storage_config)
+
+            # 2nd case: Execute in Background as a subprocess. The main program does not wait for its completion.
+            storage_config = json.dumps(self.storage_handler.get_storage_config())
+            storage_config = storage_config.replace('"', '\\"')
+
+            cmdstr = ("python3 -c 'from pywren_ibm_cloud.storage.cleaner import clean_bucket; \
```

Comment:
@JosepSampe some doesn't have python3, only python. Why we can't use Python code only?

Reply:
@gilv I changed the python interpreter used to run the data_cleaner in commit 5bca99a